### DISCUSSION TITLE
Graphin 2.8 版本设计

### DISCUSSION THREAD

**Discussion by pomelo-nwu** [antvis/Graphin#490] [Ideas]
G6@5.0 的 Alpha 版本已经出炉，Graphin 将围绕新的 G6@5.0 做一次大升级，考虑到上层业务的使用，这次版本升级，希望不再有BreakChange，暂定还是 2.8 版本

首先，AntV 图技术栈的三个产品：G6，Graphin，G6VP 为什么要这么设计？这里面有一篇[文章介绍](https://www.yuque.com/antv/gi/hl7elg)

- G6：解决了关系数据 从「数据描述」 到 「视图展示」这个过程中的的渲染 ， 布局，交互等问题，并且提供自定义机制，插件体系与丰富的算法能力。它是纯 JS 引擎
- Graphin：解决了工程应用中 框架绑定，数据驱动，分析能力组件化的问题。它是组件库。
- G6VP：解决了用户如何在线分析图数据，开发者如何快速构建图应用。它有设计规范，产品规范，是 PaaS 平台产品。

目前能想到下个版本的重要升级是：

- 工程改造：支持 react 18，支持 useGraphin hook，改为 function component 写法
- 升级 G6@5.0：支持最新的 G6 特性

**Comment by pomelo-nwu** [comment]
@Aarebecca @Yanyan-Wang 可以在这里讨论

**Comment by pomelo-nwu** [comment]
目前 Graphin 不支持 React 18：https://github.com/antvis/Graphin/issues/511

**Comment by copsman** [comment]
I love Graphin,

Can we get an estimated date when we will be able to use G6 with Graphin?

thanks!
